### PR TITLE
55 metadata products list

### PR DESCRIPTION
**Pull request by DutchBen**
- Searchfield is emitting an alert until full ES compatible search is enabled on the backend. The Subscriptions listing page is still working as it was but some dirty hacks where employed to make this both work (and not work)

- I have made a small beginning with renaming some badge components that I use with the WFO prefix.

### CODE REVIEW

File: apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx [hunk 1]
```diff
@@ -79,10 +83,27 @@ export const Products: FC<ProductsProps> = ({
...
             name: COLUMN_LABEL_STATUS,
+            width: '90',
+            render: (value) => (
+                <WFOStatusBadge status={value.toLocaleLowerCase()} />
+            ),
         },
         productBlocks: {
             field: PRODUCT_FIELD_PRODUCT_BLOCKS,
             name: COLUMN_LABEL_PRODUCT_BLOCKS,
+            render: (productBlocks) => {
```
Comment (tjeerddie) [inline apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx:94]: you don't need the function brackets when only having a return inside `(productBlocks) => (<></>)`. same with the function inside of this.

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx [hunk 2]
```diff
@@ -235,30 +229,24 @@ export const Subscriptions: FC<SubscriptionsProps> = ({
...
             pagination={pagination}
-            filterQuery={filterQuery}
-            isInvalidFilterQuery={queryContainsInvalidParts}
             isLoading={isFetching}
             localStorageKey={SUBSCRIPTIONS_TABLE_LOCAL_STORAGE_KEY}
-            onUpdateFilterQuery={setFilterQuery}
-            onCriteriaChange={handleCriteriaChange}
-            onUpdatePageSize={setPageSize}
-            onResetPageSize={() => setPageSize(defaultPageSize)}
+            onUpdatePageSize={(page) => {
```
Comment (tjeerddie) [inline apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx:245]: move this to a const function

File: libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx [hunk 3]
```diff
@@ -135,7 +142,11 @@ export const TableWithFilter = <T,>({
                 onDataSort={onUpdateDataSort}
                 pagination={pagination}
                 isLoading={isLoading}
-                onCriteriaChange={onCriteriaChange}
+                onCriteriaChange={(criterion: Criteria<T>) => {
```
Comment (tjeerddie) [inline libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx:145]: move this to a const function

File: libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx [hunk 4]
```diff
@@ -22,57 +22,57 @@ import {
...
-    isInvalidFilterQuery: boolean;
+    esQueryString?: string;
     isLoading: boolean;
     localStorageKey: string;
-    onUpdateFilterQuery: (updatedFilterQuery: string) => void;
-    onCriteriaChange: ({ page }: Criteria<T>) => void;
-    onUpdatePageSize: (updatedPageSize: number) => void;
-    onResetPageSize: () => void;
+    onUpdateEsQueryString: (esQueryString: string) => void;
+    onUpdatePageSize: (criterion: Criteria<T>['page']) => void;
```
Comment (ricardovdheijden) [inline libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx:46]: Suggesting to change the name "onUpdatePageSize" according to the data that is passed along. Page object contains the index and size.
Also maybe introduce an alias for Criteria<T>['page'] or even introduce a custom type without refering to Criteria

File: libs/orchestrator-ui-components/src/lib/components/StartPage/ListStartPage.tsx [hunk 5]
```diff
@@ -15,7 +15,8 @@ interface ListStartPage {
 
 export default function ListStartPage({ list }: ListStartPage): ReactElement {
     return (
-        list && (
+        list &&
+        list.items && (
```
Comment (ricardovdheijden) [inline libs/orchestrator-ui-components/src/lib/components/StartPage/ListStartPage.tsx:19]: Could shorten this to list?.items

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx [hunk 6]
```diff
@@ -235,30 +229,24 @@ export const Subscriptions: FC<SubscriptionsProps> = ({
...
-        if (page) {
-            const { index, size } = page;
-            setPageSize(size);
-            setPageIndex(index * size);
-        }
-    };
-
     return (
-        <TableWithFilter
+        <TableWithFilter<Subscription>
```
Comment (ricardovdheijden) [inline apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx:233]: Adding the type is not needed I would say. TS can determine the type based on de props

File: apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx [hunk 7]
```diff
@@ -122,24 +142,24 @@ export const Products: FC<ProductsProps> = ({
...
-                    setDataDisplayParam,
-                )}
-                pagination={pagination}
-                isLoading={isFetching}
-                onCriteriaChange={getPageSizeHandler<Product>(
-                    setDataDisplayParam,
-                )}
-            />
-        </>
+        <TableWithFilter<Product>
```
Comment (ricardovdheijden) [inline apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx:145]: Providing the type Product here is not needed, I would say.

File: apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx (hunk 1 above)
Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx:94]: It's my personal preference to always include explicit returns. If you have to add something later - or temporarily like a console.log - it's easier. I'm as flexible though but I think we should probably make it a rule: 
https://eslint.org/docs/latest/rules/arrow-body-style, "as-needed" ?

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx (hunk 2 above)
Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx:245]: What do you mean with a const function?

File: libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx [hunk 8]
```diff
@@ -102,24 +102,31 @@ export const TableWithFilter = <T,>({
...
+            size: updatedTableConfig.selectedPageSize,
+        });
     };
 
     const handleResetToDefaults = () => {
         setHiddenColumns(defaultHiddenColumns);
         setShowSettingsModal(false);
         clearTableConfigFromLocalStorage(localStorageKey);
-        onResetPageSize();
+        onUpdatePageSize({
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx:115]: I have fixed it so it remembers the initial pageSize passed into it and reuses that

File: libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx (hunk 4 above)
Comment (DutchBen) [reply libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx:46]: Updated the name of the function to onUpdatePage. Creating an alias is trickier with the generic being passed in, I think it's still nice to extend from the Criteria since it's supplied by EUI.

File: libs/orchestrator-ui-components/src/lib/components/StartPage/ListStartPage.tsx (hunk 5 above)
Comment (DutchBen) [reply libs/orchestrator-ui-components/src/lib/components/StartPage/ListStartPage.tsx:19]: fixed

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx (hunk 6 above)
Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx:233]: Can't hurt either right?

File: apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx (hunk 7 above)
Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx:145]: Can't hurt either right?

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx (hunk 2 above)
Comment (tjeerddie) [reply apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx]: what I mean is to move it out of the return and into a const above it.
```
const updatePageSize => {
  if (page) {
     setPageSize(page.size);
  }
}

return (<TableWithFilter onUpdatePageSize={updatePageSize} />
```

File: apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx (hunk 1 above)
Comment (tjeerddie) [reply apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx:94]: Good to make it a rule and you can always add the brackets when it becomes more then just a return.

Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx]: made the rule and enforced it

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx (hunk 2 above)
Comment (DutchBen) [reply apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx]: fixed

File: libs/orchestrator-ui-components/src/lib/components/SearchBar/SearchField.tsx [hunk 9]
```diff
@@ -1,31 +1,48 @@
...
+
+    return (
+        <EuiFormRow
+            fullWidth
+            isInvalid={!queryIsValid}
+            error={['The query contains invalid parts']}
+        >
+            <EuiSearchBar
+                query={queryString}
+                onChange={({ queryText }) => {
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply libs/orchestrator-ui-components/src/lib/components/SearchBar/SearchField.tsx:27]: fixed

File: libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx (hunk 3 above)
Comment (DutchBen) [reply libs/orchestrator-ui-components/src/lib/components/Table/TableWithFilter/TableWithFilter.tsx:146]: fixed

File: apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx (hunk 6 above)
Comment (ricardovdheijden) [reply apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx:239]: True, but seeing this for example in plain text there might be a confusion if this is a <Subscription> component. Also I always understood that we should let typescript do its type-inference-thing as much as possible and only add types when TS is getting confused or when it is unclear.

**Review by ricardovdheijden — APPROVED** [review]
